### PR TITLE
Absorb diego

### PR DESCRIPTION
**Pull request by jmcarp**
There are a few things we need to discuss about https://favro.com/card/1e11108a2da81e3bd7153a7a/18F-3508, so this isn't complete. To be determined:

* Should we switch from `spiff` to `spruce` for the cf deployment? Note: we can't/won't switch diego, since we're still using the upstream manifest generation script.
* Should we consolidate everything in `cg-deploy-diego` to this repo or continue to pull it in?

### CODE REVIEW

**Comment by LinuxBozo** [comment]
My thoughts are we should probably go ahead and deprecate `cg-deploy-diego` and just pull everything in here, and refactor as needed. As for spruce, doesn't seem worth the effort at this point, given we still have to deal with spiff for diego, and eventually there will be an upgrade path to `cf-deployment` and we'll want to jump ship to that when ready.

File: diego/update-isolation-segments.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
+
+set -eux
+
+cf api "${CF_API_URL}"
```
Comment (cnelson) [inline diego/update-isolation-segments.sh]: WDYT about putting this into the terraform now that it's all in one repo?

File: diego/update-isolation-segments.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+(set +x; cf auth "${CF_USERNAME}" "${CF_PASSWORD}")
+
+cf target -o "${PLATFORM_ORGANIZATION}"
+
+cf create-isolation-segment "${PLATFORM_ISOLATION_SEGMENT}"
```
Comment (cnelson) [inline diego/update-isolation-segments.sh]: WDYT about putting this into terraform now that this is all in one repo?

File: pipeline.yml [hunk 3]
```diff
@@ -773,7 +903,6 @@ jobs:
   - aggregate:
     - get: terraform-templates
       resource: cf-manifests-production
-      passed: [deploy-cf-production]
```
Comment (cnelson) [inline pipeline.yml]: If we are going to dropped the passed on these WDYT about using a separate git resource that only triggers on terraform/ path?

Comment (jmcarp) [inline pipeline.yml]: SGTM, updated.

File: diego/update-isolation-segments.sh (hunk 2 above)
Comment (jmcarp) [inline diego/update-isolation-segments.sh]: I think we'd want to PR https://github.com/orange-cloudfoundry/terraform-provider-cloudfoundry to allow setting space isolation segments first.

**Pull request merged**